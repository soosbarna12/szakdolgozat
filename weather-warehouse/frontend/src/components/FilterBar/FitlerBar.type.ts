import { Dayjs } from 'dayjs';
import { Pages } from '../../types/page.type';

export interface FilterBarProps {
  type: Pages;
  location: string;
  onDateChange?: (dateValue: Dayjs | null) => void;
  onSaveLocation?: () => void;
  onExportLocation?: () => void;
  onResetLocation?: () => void;

}
