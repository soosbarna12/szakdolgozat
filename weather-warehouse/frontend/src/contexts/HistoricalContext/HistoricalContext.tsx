import React, { createContext, useEffect, useState } from "react";
import { LOCAL_STORAGE_TEMPERATURE_SCALE } from "../../consts/temperatureScale.const";
import { TemperatureScale } from "../../types/temperatureScale.type";
import { HistoricalContextType, HistoricalLocationData, HistoricalPageData, HistoricalProviderProps } from "./HistoricalContext.type";


export const HistoricalContext = createContext<HistoricalContextType>({
  location: { name: "", lat: 0, lon: 0 },
  setLocation: () => { },
  historicalPageData: [] as any,
  setHistoricalPageData: () => { },
  temperatureScale: TemperatureScale.Celsius,
  setTemperatureScale: () => { },
});

export const HistoricalLocationProvider = ({ children }: HistoricalProviderProps) => {

  const [location, setLocation] = useState<HistoricalLocationData>(() => {
    const storedLocation = localStorage.getItem("historicalLocation");
    const defaultLocation = { name: "", lat: 0, lon: 0 };
    try {
      return storedLocation ? JSON.parse(storedLocation) : defaultLocation;
    } catch (error) {
      console.error("Failed to parse location from localStorage:", error);
      return defaultLocation;
    }
  });

  const [historicalPageData, setHistoricalPageData] = useState<HistoricalPageData[]>(() => {
    const storedData = localStorage.getItem("historicalPageData");
    try {
      return storedData ? JSON.parse(storedData) : [];
    } catch (error) {
      console.error("Failed to parse historical data from localStorage:", error);
      return [];
    }
  });

  const [temperatureScale, setTemperatureScale] = useState<TemperatureScale | string>(() => {
    const storedTemperatureScale = localStorage.getItem(LOCAL_STORAGE_TEMPERATURE_SCALE);
    try {
      return storedTemperatureScale ? JSON.parse(storedTemperatureScale) : TemperatureScale.Celsius;
    } catch (error) {
      console.error("Failed to parse temperature scale from localStorage:", error);
      return TemperatureScale.Celsius;
    }
  });


  useEffect(() => {
    if (historicalPageData) {
      localStorage.setItem("historicalPageData", JSON.stringify(historicalPageData));
    }
  }, [historicalPageData]);

  useEffect(() => {
    localStorage.setItem("historicalLocation", JSON.stringify(location));
  }, [location]);

  useEffect(() => {
    localStorage.setItem(LOCAL_STORAGE_TEMPERATURE_SCALE, JSON.stringify(temperatureScale));
  }, [temperatureScale]);


  return (
    <HistoricalContext.Provider value={{ location, setLocation, historicalPageData, setHistoricalPageData, temperatureScale, setTemperatureScale }}>
      {children}
    </HistoricalContext.Provider>
  );
};
