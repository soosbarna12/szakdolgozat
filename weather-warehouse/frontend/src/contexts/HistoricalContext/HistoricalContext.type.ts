import { Dispatch, ReactNode, SetStateAction } from "react";
import { HistoricalDataTable } from "../../types/historicalDataTable.type";
import { TemperatureScale } from "../../types/temperatureScale.type";

export interface HistoricalLocationData {
  name: string;
  country?: string;
  lat: number;
  lon: number;
}

// eslint-disable-next-line @typescript-eslint/no-empty-object-type
export interface HistoricalPageData extends HistoricalDataTable {}

export interface HistoricalContextType {
  location: HistoricalLocationData;
  historicalPageData: any [];
  temperatureScale: TemperatureScale | string;
  setLocation: Dispatch<SetStateAction<HistoricalLocationData>>;
  setHistoricalPageData: Dispatch<SetStateAction<HistoricalPageData[]>>;
  setTemperatureScale: Dispatch<SetStateAction<TemperatureScale | string>>;
}

export interface HistoricalProviderProps {
  children: ReactNode;
}
