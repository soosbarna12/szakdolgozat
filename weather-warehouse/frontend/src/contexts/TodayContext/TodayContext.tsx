import React, { createContext, useEffect, useState } from "react";
import { TodayContextType, TodayLocationData, TodayProviderProps } from "./TodayContext.type";


export const TodayContext = createContext<TodayContextType>({
	location: { name: "", lat: 0, lon: 0 },
	setLocation: () => { },
});

export const TodayLocationProvider = ({ children }: TodayProviderProps) => {

	const [location, setLocation] = useState<TodayLocationData>(() => {
		const storedLocation = localStorage.getItem("todayLocation");
		const defaultLocation = { name: "", lat: 0, lon: 0 };
		try {
			return storedLocation ? JSON.parse(storedLocation) : defaultLocation;
		} catch (error) {
			return defaultLocation;
		}
	});

	useEffect(() => {
		localStorage.setItem("todayLocation", JSON.stringify(location));
	}, [location]);


	return (
		<TodayContext.Provider value={{ location, setLocation }}>
			{children}
		</TodayContext.Provider>
	);
};
