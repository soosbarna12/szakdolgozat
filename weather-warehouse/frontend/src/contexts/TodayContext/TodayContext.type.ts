import { Dispatch, ReactNode, SetStateAction } from "react";

// location name interfere
export interface TodayLocationData {
  name: string;
  country?: string;
  state?: string;
  lat: number;
  lon: number;
}

export interface TodayContextType {
  location: TodayLocationData;
  setLocation: Dispatch<SetStateAction<TodayLocationData>>;
}

export interface TodayProviderProps {
	children: ReactNode;
}
