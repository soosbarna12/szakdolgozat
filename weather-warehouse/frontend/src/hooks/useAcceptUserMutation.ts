import { useMutation } from "@tanstack/react-query";
import { useAlert } from "../utils/AlertContext";
import axios from "../utils/axiosConfig";


export function useAcceptUserMutation() {
  const { showAlert } = useAlert();

  const mutation = useMutation({
    mutationFn: async (id: number) => {
      const response = await axios.post(`/user/accept`, { id });
      return response.data;
    },
    onSuccess: () => {
      showAlert("User accepted successfully", "success");
    },
    onError: () => {
      showAlert("Failed to accept user", "error");
    },
  });

  return mutation;
}
