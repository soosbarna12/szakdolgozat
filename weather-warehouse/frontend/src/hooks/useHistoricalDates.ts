import { useQuery } from "@tanstack/react-query";
import dayjs from "dayjs";
import { useEffect } from "react";
import { useAlert } from "../utils/AlertContext";
import axios from "../utils/axiosConfig";

export function useHistoricalDates(location: string) {
  const { showAlert } = useAlert();

  const { data, error, isLoading } = useQuery({
    queryKey: ["historicalDates", location],
    queryFn: async () => {
      const response = await axios.get(`/api/historical/historicalDates`, {
        params: { location },
      });
      
      return response.data.map((date: any) => dayjs(date.date).format("YYYY-MM-DD"));
    },
    
    enabled: !!(location),
    refetchOnWindowFocus: false,
  });

  useEffect(() => {
    if (error) {
      showAlert("Error fetching historical dates", "error");
    }
  }, [error]);

  return { data: data || [], error, isLoading };
}
