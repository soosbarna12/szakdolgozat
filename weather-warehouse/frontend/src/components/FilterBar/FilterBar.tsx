import { AppBar, CssBaseline, Toolbar } from "@mui/material";
import React from "react";
import { HistoricalContext } from "../../contexts/HistoricalContext/HistoricalContext";
import { TodayContext } from "../../contexts/TodayContext/TodayContext";
import { Pages } from "../../types/page.type";
import { ActionsButton } from "./Actions/ActionsButton/ActionsButton";
import { DateFilter } from "./DateFilter/DateFilter";
import { FilterBarProps } from "./FitlerBar.type";
import { LocationSearch } from "./LocationSearch/LocationSearch";


export function FilterBar(props: Readonly<FilterBarProps>) {
  const { type, onDateChange, onExportLocation, onSaveLocation, onResetLocation } = props;
  const { location: historicalLocation, setLocation: historicalSetLocation } = React.useContext(HistoricalContext);
  const { location: todayLocation, setLocation: todaySetLocation } = React.useContext(TodayContext);

  function renderFilterBar() {
    if (type === Pages.Historical) {
      return (
        <>
          <LocationSearch type={Pages.Historical} location={historicalLocation} setLocation={historicalSetLocation} />
          <DateFilter onDateChange={onDateChange} />
          <ActionsButton onSaveLocation={onSaveLocation} onExportLocation={onExportLocation} onResetLocation={onResetLocation} />
        </>
      );
    }

    return (
      <LocationSearch type={type} location={todayLocation} setLocation={todaySetLocation} />
    );
  }

  return (
    <>
      <CssBaseline />
      <AppBar position="static" color="transparent" elevation={0}>
        <Toolbar sx={{ justifyContent: 'center', padding: 0, marginBottom: 1 }}>
          {renderFilterBar()}
        </Toolbar>
      </AppBar>
    </>
  );
}
