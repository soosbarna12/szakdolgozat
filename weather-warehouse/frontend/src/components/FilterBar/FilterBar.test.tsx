import '@testing-library/jest-dom';
import { fireEvent, screen, waitFor } from '@testing-library/react';
import React from 'react';
import { HistoricalContext } from '../../contexts/HistoricalContext/HistoricalContext';
import { TodayContext } from '../../contexts/TodayContext/TodayContext';
import { useHistoricalDates } from '../../hooks/useHistoricalDates';
import { Pages } from '../../types/page.type';
import { renderWithQueryClient } from '../../utils/test/renderWithQueryClient';
import { FilterBar } from './FilterBar';


jest.mock('../../hooks/useHistoricalDates', () => ({
    useHistoricalDates: jest.fn(),
}));

describe('FilterBar/FilterBar', () => {
    const mockOnDateChange = jest.fn();
    const mockOnExportLocation = jest.fn();
    const mockOnSaveLocation = jest.fn();
    const mockOnResetLocation = jest.fn();
    const mockHistoricalDates = ['2023-01-01', '2023-01-02', '2023-01-03'];

    const historicalContextValue = {
        location: { name: 'Historical Location', lat: 40.7128, lon: -74.006 },
        setLocation: jest.fn(),
    };

    const todayContextValue = {
        location: { name: 'Today Location', lat: 34.0522, lon: -118.2437 },
        setLocation: jest.fn(),
    };

    const defaultProps = {
        type: Pages.Historical,
        onDateChange: mockOnDateChange,
        onExportLocation: mockOnExportLocation,
        onSaveLocation: mockOnSaveLocation,
        onResetLocation: mockOnResetLocation,
    };

    beforeEach(() => {
        jest.clearAllMocks();
        (useHistoricalDates as jest.Mock).mockImplementation(() => ({
            data: mockHistoricalDates,
            isLoading: false,
        }));

        // Mock localStorage
        Object.defineProperty(window, 'localStorage', {
            value: {
                getItem: jest.fn((key) => {
                    if (key === 'token') {
                        return "23456134123";
                    }
                    return null;
                }),
                setItem: jest.fn(),
            },
            writable: true,
        });
    });

    function renderWithHistoricalContext() {
        return renderWithQueryClient(
            <HistoricalContext.Provider value={historicalContextValue as any}>
                <TodayContext.Provider value={todayContextValue}>
                    <FilterBar {...defaultProps as any} />
                </TodayContext.Provider>
            </HistoricalContext.Provider>
        );
    }

    function renderWithTodayContext() {
        return renderWithQueryClient(
            <HistoricalContext.Provider value={historicalContextValue as any}>
                <TodayContext.Provider value={todayContextValue}>
                    <FilterBar {...defaultProps as any} type={Pages.Today} />
                </TodayContext.Provider>
            </HistoricalContext.Provider>
        );
    }

    it('matches FilterBar component snapshot', () => {
        const component = renderWithHistoricalContext();
        expect(component).toMatchSnapshot();
    });

    it('renders historical filter bar when type is Historical', () => {
        renderWithHistoricalContext();

        expect(screen.getByTestId('locationSearchBox')).toBeInTheDocument();
        expect(screen.getByPlaceholderText('Date')).toBeInTheDocument();
        expect(screen.getByTestId('actionsButton')).toBeInTheDocument();
    });

    it('renders today filter bar when type is Today', () => {
        renderWithTodayContext();

        expect(screen.getByTestId('locationSearchBox')).toBeInTheDocument();
        expect(screen.queryByPlaceholderText('Date')).not.toBeInTheDocument();
        expect(screen.queryByTestId('actionsButton')).not.toBeInTheDocument();
    });

    it('calls onSaveLocation when save button is clicked', async () => {
        renderWithHistoricalContext();

        const actionsButton = screen.getByTestId('actionsButton');
        fireEvent.click(actionsButton);
    
        // Verify that the ActionsMenu is rendered
        const saveButton = await screen.findByTestId('saveMenuItem');
        fireEvent.click(saveButton);

        await waitFor(() => {
            expect(mockOnSaveLocation).toHaveBeenCalled();
        });
    });

    it('calls onExportLocation when export button is clicked', async () => {
        renderWithHistoricalContext();

        const actionsButton = screen.getByTestId('actionsButton');
        fireEvent.click(actionsButton);
    
        // Verify that the ActionsMenu is rendered
        const exportButton = await screen.findByTestId('exportMenuItem');
        fireEvent.click(exportButton);

        await waitFor(() => {
            expect(mockOnExportLocation).toHaveBeenCalled();
        });
    });

    it('calls onResetLocation when reset button is clicked', async () => {
        renderWithHistoricalContext();

        const actionsButton = screen.getByTestId('actionsButton');
        fireEvent.click(actionsButton);
    
        // Verify that the ActionsMenu is rendered
        const resetButton = await screen.findByTestId('resetMenuItem');
        fireEvent.click(resetButton);

        await waitFor(() => {
            expect(mockOnResetLocation).toHaveBeenCalled();
        });
    });
});