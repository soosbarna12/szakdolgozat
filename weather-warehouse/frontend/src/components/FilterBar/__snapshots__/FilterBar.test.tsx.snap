// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBar/FilterBar matches FilterBar component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionStatic css-94iixj-MuiPaper-root-MuiAppBar-root"
        style="--Paper-shadow: none;"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-t4zkma-MuiToolbar-root"
        >
          <div
            class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-1tlcqt-MuiAutocomplete-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1hrwdwy-MuiFormControl-root-MuiTextField-root"
              data-testid="locationSearchBox"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-1e3qgpi-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1nowbqt-MuiInputAdornment-root"
                >
                  <span
                    aria-hidden="true"
                    class="notranslate"
                  >
                    ​
                  </span>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                    data-testid="SearchOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
                    />
                  </svg>
                </div>
                <input
                  aria-autocomplete="list"
                  aria-expanded="false"
                  aria-invalid="false"
                  autocapitalize="none"
                  autocomplete="off"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-qve9l1-MuiInputBase-input-MuiOutlinedInput-input"
                  id="«r0»"
                  placeholder="Location"
                  role="combobox"
                  spellcheck="false"
                  type="text"
                  value=""
                />
                <div
                  class="MuiAutocomplete-endAdornment css-1uhhrmm-MuiAutocomplete-endAdornment"
                >
                  <button
                    aria-label="Open"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-2y136s-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                    tabindex="-1"
                    title="Open"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="ArrowDropDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </button>
                </div>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-w4cd9x"
                  >
                    <span
                      aria-hidden="true"
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root css-qkn3r8-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-jupps9-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                autocomplete="off"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-13meb6w-MuiInputBase-input-MuiOutlinedInput-input"
                id="«r3»"
                inputmode="text"
                placeholder="Date"
                type="text"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-elo8k2-MuiInputAdornment-root"
              >
                <button
                  aria-label="Choose date"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1ysp02-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                    data-testid="CalendarIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                    />
                  </svg>
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-w4cd9x"
                >
                  <span
                    aria-hidden="true"
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiBox-root css-2uchni"
            data-testid="actionsButton"
          >
            <button
              aria-label="Open actions"
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-16j2zoz-MuiButtonBase-root-MuiButton-root"
              data-mui-internal-clone-element="true"
              tabindex="0"
              type="button"
            >
              Actions
              <span
                class="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeMedium css-1gulhci-MuiButton-endIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="KeyboardArrowDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </header>
    </div>
    <div
      aria-hidden="true"
      class="MuiPopover-root MuiMenu-root MuiModal-root MuiModal-hidden css-14cd561-MuiModal-root-MuiPopover-root-MuiMenu-root"
      id="menu-appbar"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiBackdrop-invisible MuiModal-backdrop css-vl6bb4-MuiBackdrop-root-MuiModal-backdrop"
        style="opacity: 0; visibility: hidden;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="-1"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiPopover-paper MuiMenu-paper css-t3bg0k-MuiPaper-root-MuiPopover-paper-MuiMenu-paper"
        style="--Paper-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12); opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden;"
        tabindex="-1"
      >
        <ul
          class="MuiList-root MuiList-padding MuiMenu-list css-1toxriw-MuiList-root-MuiMenu-list"
          role="menu"
          tabindex="-1"
        >
          <li
            class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-nl63xq-MuiButtonBase-root-MuiMenuItem-root"
            data-testid="saveMenuItem"
            role="menuitem"
            tabindex="0"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1xrrpwr-MuiTypography-root"
            >
              Save
            </p>
          </li>
          <li
            class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-nl63xq-MuiButtonBase-root-MuiMenuItem-root"
            data-testid="exportMenuItem"
            role="menuitem"
            tabindex="-1"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1xrrpwr-MuiTypography-root"
            >
              Export
            </p>
          </li>
          <li
            class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-nl63xq-MuiButtonBase-root-MuiMenuItem-root"
            data-testid="resetMenuItem"
            role="menuitem"
            tabindex="-1"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1xrrpwr-MuiTypography-root"
            >
              Reset
            </p>
          </li>
        </ul>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="-1"
      />
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionStatic css-94iixj-MuiPaper-root-MuiAppBar-root"
      style="--Paper-shadow: none;"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-t4zkma-MuiToolbar-root"
      >
        <div
          class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-1tlcqt-MuiAutocomplete-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1hrwdwy-MuiFormControl-root-MuiTextField-root"
            data-testid="locationSearchBox"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-1e3qgpi-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1nowbqt-MuiInputAdornment-root"
              >
                <span
                  aria-hidden="true"
                  class="notranslate"
                >
                  ​
                </span>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="SearchOutlinedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
                  />
                </svg>
              </div>
              <input
                aria-autocomplete="list"
                aria-expanded="false"
                aria-invalid="false"
                autocapitalize="none"
                autocomplete="off"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-qve9l1-MuiInputBase-input-MuiOutlinedInput-input"
                id="«r0»"
                placeholder="Location"
                role="combobox"
                spellcheck="false"
                type="text"
                value=""
              />
              <div
                class="MuiAutocomplete-endAdornment css-1uhhrmm-MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Open"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-2y136s-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                  tabindex="-1"
                  title="Open"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-w4cd9x"
                >
                  <span
                    aria-hidden="true"
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root css-qkn3r8-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-jupps9-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-13meb6w-MuiInputBase-input-MuiOutlinedInput-input"
              id="«r3»"
              inputmode="text"
              placeholder="Date"
              type="text"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-elo8k2-MuiInputAdornment-root"
            >
              <button
                aria-label="Choose date"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1ysp02-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="CalendarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                  />
                </svg>
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-w4cd9x"
              >
                <span
                  aria-hidden="true"
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiBox-root css-2uchni"
          data-testid="actionsButton"
        >
          <button
            aria-label="Open actions"
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-16j2zoz-MuiButtonBase-root-MuiButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            Actions
            <span
              class="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeMedium css-1gulhci-MuiButton-endIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="KeyboardArrowDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
